sources/main.functions.php

```diff
@@ -2631,13 +2632,10 @@ function storeUsersShareKey(
         }
     } else {
         // Create sharekey for each user
-        //DB::debugmode(true);
         $users = DB::query(
             'SELECT id, public_key
             FROM ' . prefixTable('users') . '
-            WHERE ' . ($onlyForUser === true ? 
-                'id IN ("' . TP_USER_ID . '","' . ($apiUserId === -1 ? (int) $session->get('user-id') : $apiUserId) . '") ' : 
-                'id NOT IN ("' . OTV_USER_ID . '","' . SSH_USER_ID . '","' . API_USER_ID . '"'.($all_users_except_id === -1 ? '' : ', "'.$all_users_except_id.'"').') ') . '
+            WHERE id NOT IN ("' . OTV_USER_ID . '","' . SSH_USER_ID . '","' . API_USER_ID . '"'.($all_users_except_id === -1 ? '' : ', "'.$all_users_except_id.'"').')
```

Comment:
Can you replace $all_users_except_id concat by %i or explicit int cast?

Reply:
of course